diff --git a/src/ubx.h b/src/ubx.h
--- a/src/ubx.h
+++ b/src/ubx.h
@@ -1203,10 +1203,20 @@
 	bool cfgValsetRaw(uint32_t key_id, uint32_t value);
 
 	/** A fixed key/value pair for cfgValset(items) */
+#if defined(_MSC_VER)
+#pragma pack(push, 1)
+#endif
 	struct CfgValsetItem {
 		uint32_t key;
 		uint8_t value;
-	} __attribute__((packed));
+	}
+#if !defined(_MSC_VER)
+	__attribute__((packed))
+#endif
+	;
+#if defined(_MSC_VER)
+#pragma pack(pop)
+#endif
 
 	/**
 	 * Add a fixed list of 1-byte configuration values
